hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmNegatedPredicate.java

```diff
@@ -79,4 +79,9 @@ public void appendHqlString(StringBuilder sb) {
 	protected SqmNegatablePredicate createNegatedNode() {
 		return new SqmNegatedPredicate( this, nodeBuilder() );
 	}
+
+	@Override
+	public SqmNegatablePredicate not() {
+		return createNegatedNode();
+	}
```

Comment:
Actually, I think we should rather always create a negated node when calling `not`, so this should be changed in `AbstractNegatableSqmPredicate#not`. If people want to avoid the negation wrapper/copy, they should IMO rather use the `SqmNegatablePredicate#negate` method.

Internally, in the HQL parser we always use `negate` to avoid the copies, so since this is a JPA Criteria method, I would prefer that this always behaves spec compliant, regardless of the JPA compliance configuration.

Any thoughts @sebersole?

Reply:
> Actually, I think we should rather always create a negated node when calling `not`, so this should be changed in `AbstractNegatableSqmPredicate#not`. 

I tried that initially, but that needs then another small adaptation for ilike, since it makes that fail. Shouldn't be a big deal though

> If people want to avoid the negation wrapper/copy, they should IMO rather use the `SqmNegatablePredicate#negate` method.
> 
> Internally, in the HQL parser we always use `negate` to avoid the copies, so since this is a JPA Criteria method, I would prefer that this always behaves spec compliant, regardless of the JPA compliance configuration.
> 
> Any thoughts @sebersole?

Reply:
I agree.  JPA has its specific requirements here (copy)